src/mixins/createCachedRenderMixin.ts

```diff
@@ -121,9 +131,15 @@ const createCachedRenderMixin: CachedRenderFactory = createStateful
 					props[key] = cachedRender.listeners[key];
 				}
 				const classes: { [index: string]: boolean; } = {};
+				const widgetClasses: string[] = widgetClassesMap.get(cachedRender);
+
+				widgetClasses.forEach((c) => classes[c] = false);
+
 				if (cachedRender.classes) {
 					cachedRender.classes.forEach((c) => classes[c] = true);
+					widgetClassesMap.set(cachedRender, cachedRender.classes);
```

Comment:
The classes should persist on every render once they are declared, otherwise I think we will run into issues with maquette and I don't think this reflects this... for example current behaviour (where each line is a render):

```
[ 'foo' ] => classes: { 'foo': true }
[ 'bar' ] => classes: { 'foo': false, 'bar': true }
[ 'baz' ] => classes: { 'bar': false, 'baz': true }
[ 'foo' ] => classes: { 'baz': false, 'foo': true }
```

But I think we need:

```
[ 'foo' ] => classes: { 'foo': true }
[ 'bar' ] => classes: { 'foo': false, 'bar': true }
[ 'baz' ] => classes: { 'foo': false, 'bar': false, 'baz': true }
[ 'foo' ] => classes: { 'foo': true, 'bar': false, 'baz': false }
```